scripts-prometheus-exporter/prometheus-exporter

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
```

Comment:
Hm, I don't really want a proliferation of random script directories at the top level; I was already a bit unhappy with having both `scripts` and `scripts-before8`. It would be better to just add `prometheus-exporter` to the normal scripts directories, and then in `update.sh` (inverting the logic you added) remove it from old version directories. Or you could simply just add it everywhere, and then in the script itself test with `-f` if `Solr-exporter` is available in the filesystem, and error out with "Prometheus is not available prior to Solr 7.3" if not; then you don't need any `build.sh`/`update.sh` changes, which will keep things easier to maintain. Do that.

Looking at the contents, it seems to me you're just translating environment variables to command-line arguments (rather than say use determining some run-time information), so they need to be configured outside of the container and get passed in. At which point, what's the value of this over just providing those arguments to the command to run form your calling environment? Out of interest, what is your orchestration environment (I know OpenShift makes doing that inconvenient)?

Reply:
I like that a lot more. (Checking for the executable in the script)

Our orchestration framework is Kubernetes, and we are looking to add metrics capabilities in the [solr-operator](https://github.com/bloomberg/solr-operator). 

I think there are 2 good reasons to have an interface (the `prometheus-exporter` script) that obfuscates the base executable.
- It allows for the user to be unaware of any changes solr makes to the location of the executable, new start scripts, or small changes in the command line parameters.
- It also makes it possible to have custom builds of solr docker imitate this interface. That way users can drop in their custom images, and infrastructure that uses the official docker image will be able to interact with the custom image in the same way.

Reply:
Thanks for such a thorough review! I think I've cleaned everything up.

I'd understand if you prefer to leave out the `prometheus-exporter` script, as the only required change is adding `chmod ..../solr-exporter` in `Dockerfile-varsolr.template`. I just think the script adds an improved user experience :)